src/extension/store.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { computed, IArrayWillSplice, intercept, observable } from 'mobx'
+import { Log, Result } from 'sarif'
+import { Memento } from 'vscode'
+import { mapDistinct } from '../shared'
+import '../shared/extension'
+
+export class Store {
+	static extensionPath: string | undefined
+	static globalState: Memento
+
+	@observable.shallow logs = [] as Log[]
+	@computed get results() {
+		const runs = this.logs.map(log => log.runs).flat()
+		return runs.map(run => run.results).filter(run => run).flat() as Result[]
+	}
+	@computed get distinctArtifactNames() {
+		const fileAndUris = this.logs.map(log => [...log._distinct.entries()]).flat()
+		return mapDistinct(fileAndUris)
+	}
+
+	constructor() {
+		intercept(this.logs, objChange => {
+			const change = objChange as unknown as IArrayWillSplice<Log>
+			change.added = change.added.filter(log => this.logs.every(existing => existing._uri !== log._uri))
```

Comment:
>change.added = change.added.filter(log => this.logs.every(existing => existing._uri !== log._uri)) [](start = 3, length = 98)

This raises an interesting point which is if a consumer using the APIs gives a log URI to the SARIF extension and then it is modified on the file-system, should the SARIF extension handle this or should the caller of the API handle this.

Reply:
I added a comment to the API to roughly capture this topic.

---
In reply to: [435538028](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r435538028) [](ancestors = 435538028)